### PR TITLE
feat(stark-ui): implementation of breadcrumb component

### PR DESCRIPTION
**Pull request by Mallikki**
ISSUES CLOSED: #591

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NationalBankBelgium/stark/blob/master/CONTRIBUTING.md#-commit-message-guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
The breadcrumb component needed to be implemented 

## What is the new behavior?
The breadcrumb component is implemented

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: packages/stark-ui/src/modules/breadcrumb/components/breadcrumb.component.scss [hunk 1]
```diff
@@ -0,0 +1,52 @@
+/* ============================================================================== */
+/*  S t a r k   B r e a d c r u m b                                               */
+/* ============================================================================== */
+/* stark: src/ui/components/breadcrumb/breadcrumb.component.pcss */
+//.stark-breadcrumb {
+//  display: none;
```
Comment (SuperITMan) [inline packages/stark-ui/src/modules/breadcrumb/components/breadcrumb.component.scss]: I see in the old stark, the display was "none" by default and changed if it was desktop. Why changing this?
Does it mean we want to show this element on mobile ?

File: packages/stark-ui/src/modules/breadcrumb/components/breadcrumb.component.scss [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    margin-bottom: 20px;
+    font-size: 18px;
+    a {
+      font-size: 13px;
+      padding: 0 10px;
+    }
+  }
+}
+
+/* END stark: src/ui/components/breadcrumb/breadcrumb.component.pcss */
```
Comment (SuperITMan) [inline packages/stark-ui/src/modules/breadcrumb/components/breadcrumb.component.scss]: Path is incorrect. Could you please change it by : 
`/* END stark-ui: src/modules/breadcrumb/components/breadcrumb/breadcrumb.component.scss */`

File: packages/stark-ui/src/modules/breadcrumb/components/breadcrumb.component.scss [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+//}
+
+.stark-breadcrumb {
+  display: block;
+  a {
+    cursor: pointer;
+  }
+}
+
+@media #{$tablet-only-query} {
```
Comment (SuperITMan) [inline packages/stark-ui/src/modules/breadcrumb/components/breadcrumb.component.scss]: In the old stark, it was `@media #{$tablet-query}`. Could you please adapt this ?

File: showcase/src/app/app.component.ts [hunk 4]
```diff
@@ -1,8 +1,9 @@
 /**
  * Angular 2 decorators and services
  */
-import { Component, OnInit, ViewEncapsulation } from "@angular/core";
+import { Component, Inject, /*Inject,*/ OnInit, ViewEncapsulation } from "@angular/core";
```
Comment (SuperITMan) [inline showcase/src/app/app.component.ts]: Could you please fix those imports ?

File: showcase/src/app/demo/breadcrumb/breadcrumb.component.html [hunk 5]
```diff
@@ -0,0 +1,4 @@
+<example-viewer [extensions]="['HTML', 'TS']" filesPath="breadcrumb/breadcrumb" exampleTitle="{{ 'SHOWCASE.DEMO.BREADCRUMB.TITLE' }}">
```
Comment (SuperITMan) [inline showcase/src/app/demo/breadcrumb/breadcrumb.component.html]: Interpolation is not needed in this case. 
Could you please adapt this code like this: `exampleTitle="SHOWCASE.DEMO.BREADCRUMB.TITLE"`

File: showcase/src/app/demo/breadcrumb/breadcrumb.component.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
+"use strict";
```
Comment (SuperITMan) [inline showcase/src/app/demo/breadcrumb/breadcrumb.component.ts]: `"use strict"` is not necessary. Could you please remove it ?

File: showcase/src/app/demo/breadcrumb/breadcrumb.component.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+import { Component, Inject, OnInit } from "@angular/core";
+import { STARK_LOGGING_SERVICE, StarkLoggingService } from "@nationalbankbelgium/stark-core";
+const componentName: string = "StarkBreadcrumb";
+
+/**
+ * AAH Breadcrumb component page
```
Comment (SuperITMan) [inline showcase/src/app/demo/breadcrumb/breadcrumb.component.ts]: AAH was the old demo project. Could you please adapt this for "Demo Breadcrumb component page" ?

File: showcase/src/assets/examples/breadcrumb/breadcrumb.ts [hunk 8]
```diff
@@ -0,0 +1,22 @@
+"use strict";
```
Comment (SuperITMan) [inline showcase/src/assets/examples/breadcrumb/breadcrumb.ts]: `"use strict"` is not necessary. Could you please remove it ?

File: showcase/src/assets/examples/breadcrumb/breadcrumb.html [hunk 9]
```diff
@@ -0,0 +1 @@
+<stark-breadcrumb breadcrumbId="breadcrumb-demonstration"></stark-breadcrumb>
```
Comment (SuperITMan) [inline showcase/src/assets/examples/breadcrumb/breadcrumb.html]: `breadcrumbId` is not an input in the implementation of breadcrumb component.
Could you please provide 2 examples for this component? One with passed `breadcrumbConfig` and the other without any inputs.

File: showcase/src/assets/examples/breadcrumb/breadcrumb.ts [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+import { Component, Inject, OnInit } from "@angular/core";
+import { STARK_LOGGING_SERVICE, StarkLoggingService } from "@nationalbankbelgium/stark-core";
+const componentName: string = "StarkBreadcrumb";
+
+/**
+ * AAH Breadcrumb component page
```
Comment (SuperITMan) [inline showcase/src/assets/examples/breadcrumb/breadcrumb.ts]: AAH was the old demo project. Could you please adapt this for "Demo Breadcrumb component page" ?

File: packages/stark-ui/src/modules/breadcrumb/components/breadcrumb.component.scss (hunk 1 above)
Comment (Mallikki) [reply packages/stark-ui/src/modules/breadcrumb/components/breadcrumb.component.scss]: Sorry my bad, I've forgotten to uncomment after I've done all my tests

File: packages/stark-ui/src/modules/breadcrumb/components/breadcrumb.component.scss [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+    display: block;
+    padding-left: 8px;
+    font-size: 15px;
+    line-height: 24px;
+    margin-bottom: 0;
+    color: #fff;
+    a {
+      display: inline-block;
+      padding: 0 6px;
+      color: var(--tertiary-dark-text-color);
```
Comment (SuperITMan) [inline packages/stark-ui/src/modules/breadcrumb/components/breadcrumb.component.scss]: This variable does not exist anymore (was part of our postcss implementation), please change it for the new one.

File: packages/stark-ui/src/modules/breadcrumb/components/breadcrumb.component.scss [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+      font-style: italic;
+      text-decoration: none;
+      font-size: 12px;
+      &:hover {
+        color: var(--mat-primary);
```
Comment (SuperITMan) [inline packages/stark-ui/src/modules/breadcrumb/components/breadcrumb.component.scss]: This variable does not exist anymore (was part of our postcss implementation), please change it for the new one.

File: showcase/src/app/demo/breadcrumb/breadcrumb.component.ts [hunk 13]
```diff
@@ -0,0 +1,51 @@
+import { Component, Inject, OnInit } from "@angular/core";
+import { STARK_LOGGING_SERVICE, StarkLoggingService } from "@nationalbankbelgium/stark-core";
+import { StarkBreadcrumbConfig } from "@nationalbankbelgium/stark-ui";
+const componentName: string = "StarkBreadcrumb";
+
+/**
+ * Demo Breadcrumb component page
+ */
+@Component({
+	selector: "breadcrumb-component",
```
Comment (SuperITMan) [inline showcase/src/app/demo/breadcrumb/breadcrumb.component.ts]: The selector of a component should not contain "component" word.
Could you please change it for "breadcrumb" or change the component name for DemoBreadcrumbComponent and change the selector for "demo-breadcrumb"

File: packages/stark-ui/src/modules/breadcrumb/breadcrumb.module.ts [hunk 14]
```diff
@@ -0,0 +1,11 @@
+import { NgModule } from "@angular/core";
+import { BrowserModule } from "@angular/platform-browser";
```
Comment (SuperITMan) [inline packages/stark-ui/src/modules/breadcrumb/breadcrumb.module.ts]: BrowserModule should be used only by app.module. 
Could you please use "CommonModule" from "@angular/common" instead ? 😊

File: showcase/src/assets/examples/breadcrumb/breadcrumb-no-input.ts [hunk 15]
```diff
@@ -0,0 +1,21 @@
+import { Component, Inject, OnInit } from "@angular/core";
+import { STARK_LOGGING_SERVICE, StarkLoggingService } from "@nationalbankbelgium/stark-core";
+const componentName: string = "StarkBreadcrumb";
+
+/**
+ * Demo Breadcrumb with no input component page
+ */
+@Component({
+	selector: "breadcrumb-component",
```
Comment (SuperITMan) [inline showcase/src/assets/examples/breadcrumb/breadcrumb-no-input.ts]: The selector of a component should not contain "component" word.
Could you please change it for "breadcrumb" or change the component name for DemoBreadcrumbComponent and change the selector for "demo-breadcrumb"

File: showcase/src/assets/examples/breadcrumb/breadcrumb-config-input.ts [hunk 16]
```diff
@@ -0,0 +1,51 @@
+import { Component, Inject, OnInit } from "@angular/core";
+import { STARK_LOGGING_SERVICE, StarkLoggingService } from "@nationalbankbelgium/stark-core";
+import { StarkBreadcrumbConfig } from "@nationalbankbelgium/stark-ui";
+const componentName: string = "StarkBreadcrumb";
```
Comment (SuperITMan) [inline showcase/src/assets/examples/breadcrumb/breadcrumb-config-input.ts]: The `componentName` has to have the same value then the selector. Please change it for the selector value.

**Pull request merged**